### ISSUE TITLE
Produces attribute with wild card media type ignores content-negotiation

### ISSUE THREAD

**Issue by kichalla** [aspnet/Mvc#1850]
For the following action and request, was expecting an Xml based response but I see a Json response.

```c#
[Produces("*/*", Type = typeof(Person))]
public IActionResult Foo()
{
    return new ObjectResult(new Person()
    {
        Name = "mike",
        Id = 10
    });
}
```
Request:
```
GET http://localhost:49643/home/foo HTTP/1.1
User-Agent: Fiddler
Host: localhost:49643
Accept: application/xml
```

Lines of code interesting for this case:
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs#L144
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs#L212

**Comment by kichalla** [comment]
Just FYI...for an action like below:
```c#
[Produces("application/*", Type = typeof(Person))]
public IActionResult Foo()
{
    return new ObjectResult(new Person()
    {
        Name = "mike",
        Id = 10
    });
}
```
a request like below gives json response...is this expected?
```
GET http://localhost:49643/home/foo HTTP/1.1
User-Agent: Fiddler
Host: localhost:49643
Accept: application/xml
```

**Comment by kichalla** [comment]
FYI: @yishaigalatzer @rynowak @harshgMSFT

**Comment by harshgMSFT** [comment]
i'd expect
```[Produces("*/*", Type = typeof(Person))]``` 
should not be a supported scenario specifically after we have solved #1065, there should be no need for a user to specify */* and we should probably block it. 

for 
```[Produces("application/*", Type = typeof(Person))]```
I would expect content negotiation to kick in and we should fix this.

**Comment by yishaigalatzer** [comment]
Throw when `*` is provided.
